bin/prover-client/src/operators/cl_stf.rs

```diff
@@ -112,26 +132,37 @@ impl ProvingOp for ClStfOperator {
         task_id: &ProofKey,
         db: &ProofDb,
     ) -> Result<ClStfInput, ProvingTaskError> {
-        let block_id = match task_id.context() {
-            ProofContext::ClStf(id) => id,
-            _ => return Err(ProvingTaskError::InvalidInput("EvmEe".to_string())),
+        let (start_block_hash, end_block_hash) = match task_id.context() {
+            ProofContext::ClStf(start, end) => (*start, *end),
+            _ => return Err(ProvingTaskError::InvalidInput("CL_STF".to_string())),
         };
-        let block_num = self.get_slot(*block_id).await?;
-        let raw_witness: Option<Vec<u8>> = self
-            .cl_client
-            .get_cl_block_witness_raw(block_num)
-            .await
-            .map_err(|e| ProvingTaskError::RpcError(e.to_string()))?;
-        let witness = raw_witness.ok_or(ProvingTaskError::WitnessNotFound)?;
-        let (pre_state, l2_block) = borsh::from_slice(&witness)?;
+
+        let mut stf_witness_payloads = Vec::new();
+        let mut blkid = end_block_hash;
+        loop {
+            let raw_witness: Option<Vec<u8>> = self
+                .cl_client
+                .get_cl_block_witness_raw(blkid)
+                .await
+                .map_err(|e| ProvingTaskError::RpcError(e.to_string()))?;
+            let witness = raw_witness.ok_or(ProvingTaskError::WitnessNotFound)?;
+            stf_witness_payloads.push(witness);
+
+            if blkid == start_block_hash {
+                break;
+            } else {
+                blkid = self.get_prev_block_id(blkid).await?;
+            }
+        }
+        stf_witness_payloads.reverse();
```

Comment:
We need some safety hatch here so we bail out if we're trying to prove more than a couple thousand blocks instead of trying to load all the way back to genesis.  Just in the event that there's a bug and we have an invalid start ID we can identify it early.  See other comment about loading ancestors.